src/decommit.rs

```diff
@@ -94,14 +94,25 @@ impl WorldDiff {
         world: &mut dyn World<T>,
         decommit: UnpaidDecommit,
         gas: &mut u32,
-    ) -> Option<Program<T>> {
+    ) -> Option<(Program<T>, usize)> {
         if decommit.cost > *gas {
             // Unlike all other gas costs, this one is not paid if low on gas.
             return None;
         }
         *gas -= decommit.cost;
-        self.decommitted_hashes.insert(decommit.code_key, ());
-        Some(world.decommit(decommit.code_key))
+        let is_old = self
+            .decommitted_hashes
+            .insert(decommit.code_key, ())
+            .is_some();
+        let decommit = world.decommit(decommit.code_key);
+
+        let decommit_cycles = if !is_old {
+            (decommit.code_page().len() + 1) / 2
+        } else {
+            0
+        };
```

Comment:
It seems to me that this could be counted based on `decommitted_hashes` as each entry there is counted once.

External rollbacks interact with this, though. It looks like we'd need to count the cycles of the decommits that are undone by a rollback.

Reply:
Only if that becomes a hashmap where the value would be the length of the code page?

Reply:
Right, the length is not readily available.